src/lib/share/polyfill.js

```diff
@@ -0,0 +1,86 @@
+import { Platform, Share } from 'react-native'
+import { isAndroidWeb, isWeb } from '../utils/platform'
+import { noopAsync } from '../utils/async'
+
+export const isWebSharingAvailable = isWeb && !!navigator.share
+export const isSharingEmulated = 'true' === process.env.REACT_APP_EMULATE_SHARING
+
+const nativeShare = shareObject => Share.share(shareObject)
+
+// eslint-disable-next-line require-await
+const emulatedShareWeb = async ({ url }) => {
+  const result = window.confirm(`Share ${url} ?`)
+
+  if (result) {
+    return
+  }
+
+  const exception = new Error('User cancelled sharing')
+
+  exception.name = 'AbortError'
+  throw exception
+}
+
+let timeoutId
+let lastListener
+
+const removeListener = () => {
+  if (timeoutId) {
+    clearTimeout(timeoutId)
+  }
+
+  window.removeEventListener('focus', lastListener)
+  lastListener = null
+}
+
+// should be non-async to avoid possible 'non-user interaction' issues
+const androidShareWeb = shareObject =>
+  new Promise((resolve, reject) => {
+    let sharingPromise
+    const exceptionHandler = exception => {
+      const { message } = exception
+
+      // Differentiate between user 'AbortError' and internal errors.
+      // E.g. Internal error: could not connect to Web Share interface.
+      if (message.startsWith('Internal error:')) {
+        exception.name = 'InternalError'
+      }
+
+      reject(exception)
+    }
+
+    removeListener()
+    window.addEventListener(
+      'focus',
+      (lastListener = () => {
+        removeListener()
+        timeoutId = setTimeout(resolve, 100)
+      }),
+    )
+
+    try {
+      sharingPromise = nativeShare(shareObject)
+    } catch (exception) {
+      exceptionHandler(exception)
+      return
+    }
+
+    sharingPromise
+      .then(resolve)
+      .catch(exceptionHandler)
+      .finally(removeListener)
+  })
```

Comment:
Alternate implementation (immediately resolve sharing promise on Android):

```suggestion
// should be non-async to avoid possible 'non-user interaction' issues
const androidShareWeb = shareObject =>
  new Promise((resolve, reject) => {
    let sharingPromise
    const exceptionHandler = exception => {
      const { message } = exception

      // Differentiate between user 'AbortError' and internal errors.
      // E.g. Internal error: could not connect to Web Share interface.
      if (message.startsWith('Internal error:')) {
        exception.name = 'InternalError'
      }

      reject(exception)
    }

    try {
      sharingPromise = nativeShare(shareObject)
    } catch (exception) {
      exceptionHandler(exception)
      return
    }

    Promise.race([sharingPromise, delay(100)])
      .then(resolve)
      .catch(exceptionHandler)
  })
```